src/main/java/cs2103/v15_1j/jimjim/JJStorage.java

```diff
@@ -0,0 +1,92 @@
+package cs2103.v15_1j.jimjim;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.Marshaller;
+import javax.xml.bind.Unmarshaller;
+
+class JJStorage implements Storage {
+	private String fileName;
+	private File saveFile;
+	private List<TaskEvent> list;
+	
+	public JJStorage() throws IOException  {
+		fileName = "data.xml";
+		saveFile = new File(fileName);
+		initializeList();
+	}
+	
+	
+	public JJStorage(String fileName) throws IOException {
+		this.fileName = fileName;
+		saveFile = new File(fileName);
+		initializeList();
+	}
+	
+	private void initializeList() throws IOException {
```

Comment:
This is an important function that should be tested too (probably through the constructor).